lib/git/lib.rb

```diff
@@ -550,6 +550,18 @@ def remove(path = '.', opts = {})
       command('rm', arr_opts)
     end
 
+    # Takes the commit message with the options and executes the commit command
+    #
+    # accepts options:
+    #  :amend
+    #  :all
+    #  :allow_empty
+    #  :author
+    #  :date
+    #  :no_verify
+    #
+    # @param [String] message the commit message to be used
+    # @param [Array] opts the commit options to be used
```

Comment:
[YARD comes with support for annotating option hashes](https://rubydoc.info/gems/yard/file/docs/Tags.md#option):

```suggestion
    # @param [Hash] opts the commit options to be used
    # @option opts [Boolean] :amend
    # @option opts [Boolean] :all
    # @option opts [Boolean] :allow_empty
    # @option opts [String] :author
    # @option opts [String] :date
    # @option opts [Boolean] :no_verify            
```

Reply:
I agree with @olleolleolle but I am planning on adding proper yard docs across the board and will fix this then.